src/lib.rs

```diff
@@ -304,11 +304,44 @@ impl BitcoinD {
                 // to be compatible with different version, in the end we are only interested if
                 // the call is succesfull not in the returned value.
                 if client_base.call::<Value>("getblockchaininfo", &[]).is_ok() {
-                    client_base
-                        .create_wallet("default", None, None, None, None)
-                        .unwrap();
-                    break Client::new(&node_url_default, Auth::CookieFile(cookie_file.clone()))
```

Comment:
did you consider a more `ask for forgiveness` approach? I suspect it would be shorter/cleaner

like matching on `client_base.create_wallet("default", None, None, None, None)` and in case of error doing the staticdir case checks

Reply:
Ah.. I think that makes more sense and I probably took your simplification idea to extreme, and removed almost all of the code.. But I think the logic still stands and we don't necessarily need to find out the existing wallet file to try to load it up..